src/main/java/seedu/duke/commands/CommandType.java

```diff
@@ -0,0 +1,26 @@
+package seedu.duke.commands;
+
+
+public enum CommandType {
+    EXIT("exit"),
+    SHOW_RATE("show-rate");
+    private String command;
+
+    CommandType(String command){
+        this.command = command;
+    }
+    public String getCommand(){
+        return command;
+    }
+    public static CommandType get(String command) {
+        if (command == null) {
+            throw new NullPointerException("Command is null");
+        }
+        for(CommandType commandType : CommandType.values()){
+            if(commandType.getCommand().equals(command)){
+                return commandType;
+            }
+        }
+        throw new IllegalArgumentException("No command found named "+command);
+    }
+}
```

Comment:
If the commands are just capital letter and the dashes are replaced with underscore, do considering implementing it in such a manner:

```java
public enum CommandType {
    EXIT, SHOW_RATE;
}
```
## String to Enum
```java
String command = args[0].replaceAll("-", "_").toUpperCase();
CommandType type = CommandType.valueOf(command);
```
By default, the second line should already throw an error if no such enum value is found. Example can be found in the other file with the ShowRate class. You can create an utility class with generics for future enums.

## Enum To String
```java
String command = CommandType.EXIT.name().replace("_", "-").toLowerCase();
```
## Alternative
Alternatively, just use an interface.

Reply:
What is the benefit to implementing it as generic commands which seem to be less readable as opposed to the current implementation?

Reply:
I see what you mean. Maybe a generic utils package is not necessary.